### ISSUE TITLE
example binder-base-derived Dockerfile that launches xvfb?

### ISSUE THREAD

**Issue by gregcaporaso** [binder-project/binder#56]
Hi all, Thanks again for binder - it's great! 

Do you guys happen to know of a Dockerfile that is derived from `binder-base` which launches xvfb? I'm trying to use a graphics package that requires an xserver, and can't get xvfb working in Binder (though I do have it working in [travis](https://github.com/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/blob/master/.travis.yml) for the project). [Here](https://github.com/gregcaporaso/built-iab/blob/master/Dockerfile)'s my latest attempt at getting this working.

**Comment by arokem** [comment]
Does this work for you? 

https://github.com/arokem/white-matter-matters/blob/master/Dockerfile

**Comment by gregcaporaso** [comment]
It looks like this adds a xvfb magic command - could you point me to an example of where that's being used? Also, it sounds like I'd have to modify the content of my notebooks for this - is that right? I'm hoping to not have to do it (I don't have to for Travis), so this solution might not work for me.

Thanks a lot for the help!

**Comment by arokem** [comment]
Yes, you would have to modify the notebook, to use the xvfb magic.

I just created a relatively small example: https://github.com/arokem/xvfb-binder

I guess another option is to set an environment variable ('USE_XVFB') and make sure that the underlying visualization code checks to see whether to use xvfbwrapper (see here for a decorator that does that: https://github.com/nipy/dipy/blob/master/dipy/testing/decorators.py#L58). 

How do you make this work on Travis?

**Comment by gregcaporaso** [comment]
[Here's](https://github.com/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/blob/master/.travis.yml) what I'm doing with travis - basically launching the xvfb process, but that script is not available in the `binder-base` docker file so I need to either get one in there, or come up with another way to launch the process. 

Do you happen to have the Dockerfile for [binder-base-headless](https://hub.docker.com/r/andrewosh/binder-base-headless/)? I'm curious to see what that one is doing, as it sounds relevant. 

I'll post back here if I get this sorted out. Thanks again for the help!

**Comment by andrewosh** [comment]
@gregcaporaso sorry for not responding to this thread earlier. Hopefully @arokem's solution worked for you, but just in case you were still experimenting, the `binder-base-headless` image is likely not the right way to go, as that was just a small experiment of mine and will not work in the production version. 

If you're writing a custom dockerfile that needs to reference a script, that script should be available in the `home/main/notebooks` directory during the build process. Have you tried that approach? 

Not entirely related, but we're going add support for a larger variety of base images in the 1.0 release, so that should make it more straightforward to write custom Dockerfiles, as the process is pretty opaque right now.

**Comment by arokem** [comment]
@andrewosh : while we're at it, could you please point to where the base images' Dockerfiles are stored? I haven't been able to find them.

**Comment by andrewosh** [comment]
@arokem yep, the current base image is here https://github.com/binder-project/binder/blob/master/images/base/, but in the 1.0 release they're all going to be in the `binder-build-core` module at https://github.com/binder-project/binder-build-core/tree/master/images/python (docs on all that very soon)

**Comment by gregcaporaso** [comment]
Thanks for the reply @andrewosh. I think I'm good with getting a script into the image - I still haven't been able to get xvfb running though (but I'm now able to reproduce the issue and test fixes locally, so that should get me moving toward solving it). Do  you have a timeline for 1.0?
